service/src/PlatformTopo.cpp

```diff
@@ -110,13 +111,29 @@ namespace geopm
     }
 
     PlatformTopoImp::PlatformTopoImp()
-        : PlatformTopoImp("")
+        : PlatformTopoImp("", try_service_proxy())
     {
 
     }
 
-    PlatformTopoImp::PlatformTopoImp(const std::string &test_cache_file_name)
+    std::unique_ptr<ServiceProxy> PlatformTopoImp::try_service_proxy(void)
+    {
+        std::unique_ptr<ServiceProxy> result;
+        try {
+            result = ServiceProxy::make_unique();
+        }
+        catch (const Exception &ex) {
+            if (std::string(ex.what()).find("Failed to open system bus") == std::string::npos) {
```

Comment:
Is there any way to make this and the other string comparison resilient to future message changes? E.g., either extract those strings to a constant used here and wherever it is generated, or add some kind of test case that ensures the throwing scenario matches this catching scenario.

Reply:
Let's take this to an architecture meeting and discuss.

Reply:
https://github.com/geopm/geopm/issues/2965